storage/storage.h

```diff
@@ -44,7 +44,23 @@ extern const uint8_t *PIN_EMPTY;
 // Maximum number of failed unlock attempts.
 // NOTE: The PIN counter logic relies on this constant being less than or equal
 // to 16.
+#if USE_TROPIC && USE_OPTIGA
+// If both Optiga and Tropic are used, every PIN attempt requires a stretched
+// PIN slot on Optiga. This restricts the total number of PIN
+// attempts.
+#define PIN_MAX_TRIES 10
+#else
 #define PIN_MAX_TRIES 16
+#endif
+
+// The number of slots configured as stretched PINs.
+#if USE_OPTIGA
+#if USE_TROPIC
+#define OPTIGA_STRETCHED_PINS_COUNT PIN_MAX_TRIES
+#else
+#define OPTIGA_STRETCHED_PINS_COUNT 1
+#endif
+#endif
```

Comment:
This constant is not necessarily Optiga-specific. If we had only Tropic, then this would still be used. In fact it is used in `core/embed/sec/tropic/tropic.c`. So I think:
- We can rename `OPTIGA_STRETCHED_PINS_COUNT` -> `STRETCHED_PIN_COUNT`.
- Move `#if USE_OPTIGA` deeper inside, or maybe even remove the condition altogether.
```suggestion
#if USE_TROPIC
#define STRETCHED_PIN_COUNT PIN_MAX_TRIES
#else
#if USE_OPTIGA
#define STRETCHED_PIN_COUNT 1
#endif
#endif
```

Reply:
Whenever we have `<something>_count`, let's please keep `<something>` singular. I don't think this is just my OCD, because I don't see anyone suggesting the use of a plural in this case. For reference a few pages that pop up in a Google search:
- `nodeCount` https://stackoverflow.com/questions/548646/naming-a-count-variable
- `errorCount` https://google.github.io/styleguide/tsguide.html#descriptive-names
- `failureCount`, `orderCount` https://medium.com/better-programming/useful-tips-for-naming-your-variables-8139cc8d44b5
- A general English usage discussion, e.g. "body count" https://www.usingenglish.com/forum/threads/plural-or-singular-when-using-with-the-word-count.242655/
- Reddit discussion, `treeCount` https://www.reddit.com/r/readablecode/comments/21tvjj/variable_naming_count_vs_number_vs_numberof_vs_n/

The same applies to `*mac_and_destroys_count` below.

Reply:
> This constant is not necessarily Optiga-specific. If we had only Tropic, then this would still be used. In fact it is used in core/embed/sec/tropic/tropic.c.

I don't agree with that. I intended for the constant to be optiga-specific. In tropic, I use `PIN_MAX_TRIES` instead. The only occurrence of `OPTIGA_STRETCHED_PINS_COUNT` in `tropic.c` is a mistake and should be replaced with `PIN_MAX_TRIES`.

Reply:
The reason why it doesn't seem Optiga-specific to me is that we have pieces of code even in storage.c that are independent of Optiga and they still work with `OPTIGA_STRETCHED_PINS_COUNT`. For example if we had a Tropic-only device, we would still need the `pin_index` or `pin_fails` to estimate time and to know which M&D slot to access in Tropic, and for that storage.c has these two pieces of code:
```c
#if OPTIGA_STRETCHED_PINS_COUNT > 1
  if (sectrue != pin_get_fails(&pin_index)) {
```
```c
#if OPTIGA_STRETCHED_PINS_COUNT > 1
  uint32_t pin_fails = 0;
  if (sectrue != pin_get_fails(&pin_fails)) {
```
Similarly when setting up the PIN we work with an array of stretched PINs in storage.c which is declared like this:
```c
#if USE_TROPIC || USE_OPTIGA
  uint8_t stretched_pins[OPTIGA_STRETCHED_PINS_COUNT][SHA256_DIGEST_LENGTH] = {
```